Sources/Sentry/SentryTracer.m

```diff
@@ -423,13 +423,27 @@ - (nullable SentryTraceContext *)traceContext
 - (void)setMeasurement:(NSString *)name value:(NSNumber *)value
 {
     SentryMeasurementValue *measurement = [[SentryMeasurementValue alloc] initWithValue:value];
-    _measurements[name] = measurement;
+
+    [self setMeasurement:name measurement:measurement];
 }
 
 - (void)setMeasurement:(NSString *)name value:(NSNumber *)value unit:(SentryMeasurementUnit *)unit
 {
+
     SentryMeasurementValue *measurement = [[SentryMeasurementValue alloc] initWithValue:value
                                                                                    unit:unit];
+    [self setMeasurement:name measurement:measurement];
+}
+
+- (void)setMeasurement:(NSString *)name measurement:(SentryMeasurementValue *)measurement
+{
+    // Although name is nonnull we saw edge cases where it was nil and then leading to crashes. If
+    // the name is nil we can discard the measurement
+    if (name == nil) {
```

Comment:
`l`: Does this also include `NULL`, `[NSNull null]`? Does it make sense to change the `setMeasurement` signature to use `_Nonnull` for the parameters?

Reply:
We already use _Nonnull, but somehow we still sometimes get `nil` here. It could be related to Kotlin Multiplatform.

Reply:
Why do you think we should also include `NULL` and `NSNull null` here? We're not dealing with dictionaries here.

Reply:
No direct reason, just double-checking after implementing #4760. If you say it is fine, I agree with you.

Reply:
I think it's fine.